File: app/src/androidTest/java/ru/santaev/clipboardtranslator/TestTranslatorApp.java
package ru.santaev.clipboardtranslator;


import java.util.List;

import io.reactivex.Flowable;
import io.reactivex.Single;
import ru.santaev.clipboardtranslator.db.entity.Language;
import ru.santaev.clipboardtranslator.model.IDataModel;

public class TestTranslatorApp extends TranslatorApp{

    @Override
    protected IDataModel buildDataModel() {
        return new IDataModel() {
            @Override
            public Single<TranslateResponse> translate(Language origin, Language target, String text) {
                return null;
            }

            @Override
            public Flowable<List<Language>> getLanguages() {
                return null;
            }

        };
    }
}

File: app/src/androidTest/java/ru/santaev/clipboardtranslator/TestTranslatorApp.kt
package ru.santaev.clipboardtranslator

import ru.santaev.clipboardtranslator.di.AppComponent
import ru.santaev.clipboardtranslator.di.DaggerAppComponent
import ru.santaev.clipboardtranslator.di.module.AppModule


class TestTranslatorApp : TranslatorApp() {

    override fun buildComponent(): AppComponent {
        return DaggerAppComponent.builder()
                .appModule(AppModule(this, true))
                .build()
    }
}
